format-common/src/main/java/io/cdap/plugin/format/plugin/AbstractFileSinkConfig.java

```diff
@@ -84,13 +84,30 @@ public void validate(FailureCollector collector) {
           .withConfigProperty(NAME_SUFFIX).withStacktrace(e.getStackTrace());
       }
     }
+
+    FileFormat format = FileFormat.JSON;
     try {
-      getFormat();
+      format = getFormat();
     } catch (IllegalArgumentException e) {
       collector.addFailure(e.getMessage(), null).withConfigProperty(NAME_FORMAT).withStacktrace(e.getStackTrace());
     }
     try {
-      getSchema();
+      Schema schema = getSchema();
+      // Checks if the output schema has fields that are not simple type.
+      // If there are non-simple field types, then serializing as CSV might not be appropriate.
+      // This restriction is on Csv, Delimited and Tab separated files.
+      if (format == FileFormat.CSV || format == FileFormat.DELIMITED || format == FileFormat.TSV) {
```

Comment:
this validation should be done in the format (DelimitedOutputFormatProvider's validate method), not in the file sink.

Reply:
discussed offline, the problem is there isn't a way to pass down the name of the format field, since we want to highlight it. So we can keep it for now and open a jira to fix this in a cleaner way.